hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineClose.java

```diff
@@ -242,7 +238,7 @@ public void testPipelineCloseWithLogFailure() throws IOException {
     xceiverRatis.handleNodeLogFailure(groupId, null);
 
     // verify SCM receives a pipeline action report "immediately"
-    Mockito.verify(pipelineActionTest, Mockito.timeout(100))
+    Mockito.verify(pipelineActionTest, Mockito.timeout(500))
```

Comment:
If it is still failing, I wonder does the timeout need to be even higher. In the init() method of this test class, I see:

```
  public void init() throws Exception {
    conf = new OzoneConfiguration();
    cluster = MiniOzoneCluster.newBuilder(conf).setNumDatanodes(3).build();
    conf.setTimeDuration(HddsConfigKeys.HDDS_HEARTBEAT_INTERVAL, 1000,
        TimeUnit.MILLISECONDS);
    pipelineDestroyTimeoutInMillis = 1000;
    conf.setTimeDuration(ScmConfigKeys.OZONE_SCM_PIPELINE_DESTROY_TIMEOUT,
        pipelineDestroyTimeoutInMillis, TimeUnit.MILLISECONDS);
   ...
```

Should the conf not be set before the MiniOzoneCluster.build() method is called? Could the cluster potentially start without the conf changes?


I cannot remember how this works. The DNs need to send a command to SCM that the pipeline should be closed here - does the DN do that over the DN heartbeat, or via some other mechanism? If the heartbeat is set to 1 second, then could this message take just over a second to arrive in the worse case?

Reply:
In `SCMDatanodeHeartbeatDispatcher.dispatch()` it seems to deal with the heartbeats, and within it, there is logic for PIPELINE_ACTIONS commands. If the commands are all sent via the heartbeat, perhaps we need a timeout of 1500. I am not sure if something on the DN can trigger the heartbeat to be send immediately if certain commands are queued on the DN though.

Reply:
I notice that there is a global Timeout variable in this class, which applies to all the methods in this class.
```
 public class TestPipelineClose {

   /**
     * Set a timeout for each test.
     */
   @Rule
   public Timeout timeout = Timeout.seconds(300);
```

Not sure if this has something to do with the failures.

Reply:
> Should the conf not be set before the MiniOzoneCluster.build() method is called?

With 1s timeout it seems to work fine both with and without that change (tested 3-400 repetitions).

https://github.com/adoroszlai/hadoop-ozone/commits/HDDS-5604-repeat (`build()` moved after conf is set)
https://github.com/adoroszlai/hadoop-ozone/commits/HDDS-5604-repro-1000 (`build()` in original order)

Reply:
Sounds like a timeout of 1000ms is the way to go then. 

I suspect the config should really all be setup before the mini-cluster is started to be sure the components get the correct config in time. Otherwise, the service could start before the config gets changed and start with the wrong value.

Reply:
> I suspect the config should really all be setup before the mini-cluster is started

Agree.

Reply:
I've updated the PR with both of these changes (setting conf and increasing timeout).